frontend/test/resolution/testIf.cpp

```diff
@@ -101,12 +101,122 @@ static void test6() {
+
+  auto& br = parseAndReportErrors(ctx, path);
+  auto mod = br.singleModule();
+  assert(mod);
+
+  assert(mod->numStmts() == 3 && mod->stmt(2)->isConditional());
+  auto cond = mod->stmt(2)->toConditional();
+  auto var = cond->condition()->toVariable();
+  assert(var);
+  auto elseCall = cond->elseStmt(0)->toFnCall();
+  assert(elseCall && elseCall->numActuals() == 1);
+  auto elseUse = elseCall->actual(0)->toIdentifier();
+  assert(elseUse);
+
+  auto& rr = resolveModule(ctx, mod->id());
+  auto& reElseUse = rr.byAst(elseUse);
+  assert(reElseUse.toId().isEmpty());
+  assert(reElseUse.type().isUnknown());
+}
+
+static void testIfVarErrorUseInElseBranchNested() {
+  Context context;
+  auto ctx = &context;
+  ErrorGuard guard(ctx);
+
+  auto path = TEST_NAME_FROM_FN_NAME(ctx);
+  std::string contents =
+    R""""(
+      class C {}
+      proc foo(x) {}
+      if var x = new C() then foo(x); else { { { foo(x); } } }
+    )"""";
+  setFileText(ctx, path, contents);
+
+  auto& br = parseAndReportErrors(ctx, path);
+  auto mod = br.singleModule();
+  assert(mod);
+
+  assert(mod->numStmts() == 3 && mod->stmt(2)->isConditional());
+  auto cond = mod->stmt(2)->toConditional();
+  auto var = cond->condition()->toVariable();
+  assert(var);
+
+  auto elseBlock2 = cond->elseStmt(0)->toBlock();
+  assert(elseBlock2 && elseBlock2->numStmts());
+  auto elseBlock3 = elseBlock2->stmt(0)->toBlock();
+  assert(elseBlock3 && elseBlock3->numStmts());
+
+  auto elseCall = elseBlock3->stmt(0)->toFnCall();
+  assert(elseCall);
+
+  assert(elseCall && elseCall->numActuals() == 1);
+  auto elseUse = elseCall->actual(0)->toIdentifier();
+  assert(elseUse);
+
+  auto& rr = resolveModule(ctx, mod->id());
+  auto& reElseUse = rr.byAst(elseUse);
+  assert(reElseUse.toId().isEmpty());
+  assert(reElseUse.type().isUnknown());
+}
+
+static void testIfVarErrorNonClassType() {
+  Context context;
+  auto ctx = &context;
+  ErrorGuard guard(ctx);
+
+  auto path = TEST_NAME_FROM_FN_NAME(ctx);
+  std::string contents =
+    R""""(
+      proc foo(x) {}
+      if var x = 1 then foo(x); else;
+    )"""";
+  setFileText(ctx, path, contents);
+
+  auto& br = parseAndReportErrors(ctx, path);
+  auto mod = br.singleModule();
+  assert(mod);
+
+  assert(mod->numStmts() == 2 && mod->stmt(1)->isConditional());
+  auto cond = mod->stmt(1)->toConditional();
+  auto var = cond->condition()->toVariable();
+  assert(var);
+
+  auto& rr = resolveModule(ctx, mod->id());
+  assert(guard.numErrors() == 1);
+  assert(guard.error(0)->message() ==
+         "a variable declared in the condition of an if statement must "
+         "be a class, not a value of type 'int(64)'");
+  auto& reVar = rr.byAst(var);
+  assert(reVar.type().kind() == QualifiedType::VAR);
+  assert(reVar.type().type() == IntType::get(ctx, 64));
+  guard.realizeErrors();
+}
+
 int main() {
   test1();
   test2();
   test3();
   test4();
   test5();
   test6();
+  testIfVarErrorUseInElseBranch();
```

Comment:
Just for consistency, I'd name these `test7` through `test9`.... or rename all other tests to be equally descriptive. Either way, I'd add plain english comments to each (new) test describing what it does (which would be a bit clearer than the camelCaseTestName).